packages/sanity/src/core/store/_legacy/document/document-pair/operations/publish.ts

```diff
@@ -33,42 +13,23 @@ export const publish: OperationImpl<[], DisabledReason> = {
     }
     return false
   },
-  execute: ({client, idPair, snapshots}) => {
-    if (!snapshots.draft) {
-      throw new Error('cannot execute "publish" when draft is missing')
-    }
-    const value = strengthenOnPublish(omit(snapshots.draft, '_updatedAt'))
-    const tx = client.observable.transaction()
-    if (!snapshots.draft) {
-      throw new Error('cannot execute "publish" when draft is missing')
-    }
-
-    if (snapshots.published) {
-      // If it exists already, we only want to update it if the revision on the remote server
-      // matches what our local state thinks it's at
-      tx.patch(idPair.publishedId, {
-        // Hack until other mutations support revision locking
-        unset: ['_revision_lock_pseudo_field_'],
-        ifRevisionID: snapshots.published._rev,
-      })
-
-      tx.createOrReplace({
-        ...value,
-        _id: idPair.publishedId,
-        _type: snapshots.draft._type,
-      })
-    } else {
-      // If the document has not been published, we want to create it - if it suddenly exists
-      // before being created, we don't want to overwrite if, instead we want to yield an error
-      tx.create({
-        ...value,
-        _id: idPair.publishedId,
-        _type: snapshots.draft._type,
-      })
-    }
-
-    tx.delete(idPair.draftId)
-
-    return tx.commit({tag: 'document.publish', visibility: 'async'})
+  execute: ({client: globalClient, idPair}) => {
+    const vXClient = globalClient.withConfig({apiVersion: 'X'})
+    const {dataset} = globalClient.config()
+
+    return vXClient.observable.request({
```

Comment:
I'm guessing we don't know which errors we'll get if the draft is deleted on the server? Is there anything else needed to ensure how this errors and throw if needed, as in the original version?

Reply:
Good point! I've asked about status on response format [here](https://linear.app/sanity/issue/CLDX-1842#comment-d763592d)